OSOE-392: Upgrade to Orchard Core 1.5

MZole opened the pull request:
Fixes #221

- https://github.com/Lombiq/Helpful-Libraries/pull/165
- https://github.com/Lombiq/Orchard-Core-API-Client/pull/10
- https://github.com/Lombiq/Orchard-Chart.js/pull/45
- https://github.com/Lombiq/Orchard-Data-Tables/pull/123
- https://github.com/Lombiq/Helpful-Extensions/pull/102
- https://github.com/Lombiq/Orchard-Azure-Application-Insights/pull/52
- https://github.com/Lombiq/Hosting-Tenants/pull/42
- https://github.com/Lombiq/Orchard-JSON-Editor/pull/30
- https://github.com/Lombiq/Orchard-Privacy/pull/41
- https://github.com/Lombiq/Orchard-Training-Demo-Module/pull/97
- https://github.com/Lombiq/Orchard-UIKit/pull/31
- https://github.com/Lombiq/Orchard-Vue.js/pull/65
- https://github.com/Lombiq/Orchard-Base-Theme/pull/44
- https://github.com/Lombiq/Hosting-Media-Theme/pull/12
- https://github.com/Lombiq/Setup-Extensions/pull/22
- https://github.com/Lombiq/UI-Testing-Toolbox/pull/225
- https://github.com/Lombiq/Audit-Trail-Extensions/pull/22
- https://github.com/Lombiq/GitHub-Actions/pull/92
- https://github.com/Lombiq/Utility-Scripts/pull/24

[OSOE-392](https://lombiq.atlassian.net/browse/OSOE-392)

MZole committed:
Updating OrchardCore NuGet packages to latest and updating submodules

MZole committed:
Updating submodules

MZole committed:
Updating submodules to new DataMigration registration

MZole committed:
Updating Lombiq.UITestingToolbox

MZole committed:
Updating Lombiq.UITestingToolbox

MZole committed:
Updating Lombiq.UITestingToolbox

MZole committed:
Updating Lombiq.UITestingToolbox

Piedone commented:
Use more descriptive commit messages than "Updating Lombiq.UITestingToolbox".

MZole committed:
Re-adding images in Lombiq.UITestingToolbox

MZole committed:
Updating HomePage images in Lombiq.UITestingToolbox

MZole committed:
Extending connection string in Lombiq.GitHub.Actions and changing to issue branch

MZole committed:
Changing back to dev branch

MZole committed:
Chagning to issue branch and updating test-dotnet script

MZole committed:
Updating OrchardCore NuGet packages in NuGet solution

MZole committed:
Updating HomePage images visual verification tests in Lombiq.UITestingToolbox

MZole committed:
Changing to dev branch on NuGet build for testing

MZole committed:
Reverting NuGet solution changes

MZole committed:
Testing IdleTenantsTests

MZole committed:
Updating submodules to latest dev branches

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Testing IdleTenantTests

MZole committed:
Updating Lombiq.Hosting.Tenants to latest dev

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Updating Lombiq.GitHub.Actions to latest dev and using OrchardCore.BackgroundTasks in UI testing toolbox

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Using const OrchardCore.Data.DatabaseProviderValue.SqlConnection in UI testing toolbox

MZole committed:
Testing if another error would raise if we don't lock the log file

MZole committed:
Updating Nlog configuration in Lombiq.UITestingToolbox

MZole committed:
Changing NLog.config and updating log file read Lombiq.UITestingToolbox

MZole committed:
Updating Lombiq.GitHub.Actions to latest dev

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

Piedone commented on .github/workflows/build-and-test.yml:
```diff
@@ -11,7 +11,7 @@ jobs:
     if: github.ref_name != github.event.repository.default_branch
     name: Build and Test - root solution (larger runners)
-    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@dev
+    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@issue/OSOE-392
```
There are no changes in the `build-and-test-orchard-core` workflow, so this and below are not needed.

Piedone commented on .github/workflows/build-and-test.yml:
```diff
@@ -41,3 +41,5 @@ jobs:
     name: Spelling
     uses: Lombiq/GitHub-Actions/.github/workflows/spelling.yml@dev
+    with:
+      spell-check-this: Lombiq/GitHub-Actions@issue/OSOE-392
```
Isn't just changing the branch reference in `uses` sufficient?

MZole replied:
I remembered correctly, it is not enough, as you can see with this [run](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/3423154830/jobs/5701423599), here I was using the issue branch.

Piedone replied:
But after merging the GHA PR, this won't be needed, right?

BTW please document [here](https://github.com/Lombiq/GitHub-Actions/blob/dev/Docs/Workflows.md#spelling-workflow) how you need this during development when updating the dictionary.

MZole replied:
Yes, after the merge this won't be needed.

This is already 'kind of' there, but I will edit that section of `spell-check-this` to make this clearer.

Piedone replied:
Thanks!

Piedone commented on src/Lombiq.OSOCE.Web/NLog.config:
```diff
@@ -14,7 +14,8 @@
         <target xsi:type="File" name="allfile" fileName="${var:configDir}/App_Data/logs/orchard-log-${shortdate}.log"
-                    layout="${longdate}|${orchard-tenant-name}|${aspnet-traceidentifier}|${event-properties:item=EventId.Id}|${logger}|${uppercase:${level}}|${message} ${exception:format=ToString,StackTrace}" />
+                    layout="${longdate}|${orchard-tenant-name}|${aspnet-traceidentifier}|${event-properties:item=EventId.Id}|${logger}|${uppercase:${level}}|${message} ${exception:format=ToString,StackTrace}"
+                    keepFileOpen="false" concurrentWrites="true" />
```
Why are these necessary?

MZole replied:
With the `Nlog` update (to 5.0) the defaults of `keepFileOpen` from false and `concurrentWrites` from true changed to the opposites and this was the root of the problem with the `IdleTenantTests`, I was getting error messages that the log file is being used by another process. (although it was only a problem on Windows, hence why it was a lot more time-consuming then I would like it to be to realise that it was the problem)
The defaults changed due to improving the performance of logging huge files, but I don't think this would benefit us much since we don't use that deep level of logging.

Piedone replied:
Can't we set this from the environment just during testing [like this](https://github.com/Lombiq/UI-Testing-Toolbox/blob/dev/Lombiq.Tests.UI.Samples/UITestBase.cs#L73) instead? Related: https://github.com/Lombiq/UI-Testing-Toolbox/issues/206

MZole replied:
We could only do this I think if we had our Nlog configuration set up from an `appsettings.json` file but we are using an `NLog.config` one. But also, don't we only use this config file for testing?

Piedone replied:
While this app is not deployed, the config is not just used for UI testing but also local development. And we'll need to figure this out now in relation to the UI Testing Toolbox, since it'll be necessary for all other apps using it too.

Piedone replied:
If we can move the NLog config to appsettings then let's do that.

MZole replied:
Turns out we can't use it from appsettings, because the `OrchardCore.Logging.NLog` requires the `NLog.config` file, and if I delete the file, I get a `FileNotFoundException` because of the line `builder.Host.UseNLogHost();` in _program.cs_ . So `OrchardCore.Logging.NLog` is also configured to use the .config file.

Piedone replied:
We could have our own extension method instead of `UseNLogHost()`, but let's just try to address this: How about adding using appsettings as a second, additional config source? So just these two properties are set from there.

snakefoot replied:
I recommend not reverting to `keepFileOpen="false"` (Instead fix the applications / tests that requires exclusive log-file-access to work. Forexample read log-files like this (And not using `File.ReadAllText()`):
```c#
using (var f = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
using (var s = new StreamReader(f))
{
    fileContent = s.ReadToEnd();
}
```
`keepFileOpen="false"` opens the door to lots of bad things. Forexample that Windows Malware Scanner suddenly can take over the file-handle, and suddenly the actual application fails to perform logging (and if not using `async="true"` then application will get stuck and completely fail). See also: https://nlog-project.org/2021/08/25/nlog-5-0-preview1-ready.html#filetarget-keepfileopen--true-by-default

Piedone replied:
Thank you for chiming in! Very valuable info. @MZole, please look into these. My suspicion is that we have the issue here: https://github.com/Lombiq/UI-Testing-Toolbox/blob/07885bbadf05a71d9d11c9856b2a7f7d649228b9/Lombiq.Tests.UI/Services/OrchardCoreInstance.cs#L132 We do exactly `File.ReadAllTextAsync()`.

Out of sheer curiosity, how did you find this convo, @snakefoot? I see you're an active contributor of NLog.

snakefoot replied:
@Piedone I regularly search github for issues about "NLog" to ensure everyone has a good time :), especially when it comes to bringing people over the major version bump.

Piedone replied:
That's awesome, thank you!

MZole committed:
Addressing change requests in submodules

MZole committed:
Updating workflow branches

MZole committed:
Updating Lombiq.GitHub.Actions to latest dev and extending docs

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Using OrchardCore.Logging.Nlog in Lombiq.UITestingToolbox

snakefoot commented on src/Lombiq.OSOCE.Web/NLog.config:
```diff
@@ -14,7 +14,8 @@
         <!-- write logs to file -->
         <target xsi:type="File" name="allfile" fileName="${var:configDir}/App_Data/logs/orchard-log-${shortdate}.log"
-                    layout="${longdate}|${orchard-tenant-name}|${aspnet-traceidentifier}|${event-properties:item=EventId.Id}|${logger}|${uppercase:${level}}|${message} ${exception:format=ToString,StackTrace}" />
+                    layout="${longdate}|${orchard-tenant-name}|${aspnet-traceidentifier}|${event-properties:item=EventId.Id}|${logger}|${uppercase:${level}}|${message} ${exception:format=ToString,StackTrace}"
```
Notice that NLog.Extensions.Logging v5 changes from `EventId_Id` + `EventId_Name` to `EventId` + `EventName`. So better to use `${event-properties:item=EventId}`. See also https://nlog-project.org/2021/08/25/nlog-5-0-preview1-ready.html#nlogextensionslogging-changes-capture-of-eventid

Piedone replied:
Thank you for this too! @MZole please also open an Orchard Core issue for this, because this needs to be fixed in OC too.

MZole committed:
Updating log file's content reading in Lombiq.UITestingToolbox

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Updating EventId in Nlog.config

Piedone commented on .github/workflows/build-and-test.yml:
```diff
@@ -11,7 +11,7 @@ jobs:
     if: github.ref_name != github.event.repository.default_branch
     name: Build and Test - root solution (larger runners)
-    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@issue/OSOE-392
+    uses: Lombiq/GitHub-Actions/.github/workflows/build-and-test-orchard-core.yml@issue/OSOE-457
```
This and below shouldn't be needed.

MZole replied:
This is from the latest dev merge, probably not intended [here](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/commit/f9901f4726b25c9e2114a81041b095633c2ca964#diff-bc668a2c9f2299cef15b222055b4b4d5311646caec2e7610e540cee18ae9b948R14).

Piedone replied:
I see. Oliver fixed them.

MZole committed:
Using ReadToEndAsync() in Lombiq.UITestingToolbox

MZole committed:
Updating Lombiq and OC NuGet package references in NuGet solution

MZole committed:
Updating OC Lucene namespace in test recipe

Piedone committed:
Re-adding future CancellationToken usage to Lombiq.UITestingToolbox

Piedone committed:
Updating Lombiq.GitHub.Actions to merged dev

Piedone committed:
Revert workflow and Lombiq.UITestingToolbox changes to see if those cause issues

Piedone committed:
Putting Spelling workflow back to dev

Piedone committed:
Refactoring in Lombiq.UITestingToolbox

Piedone committed:
Removing Obsolete types/members

Piedone committed:
Reverting refactoring in Lombiq.UITestingToolbox

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

Piedone committed:
Revert removing Obsolete types/members in /Lombiq.UITestingToolbox, because perhaps that causes the app not to start?

Piedone committed:
Revert previous

Piedone committed:
Removing just obsolete AccessbilityCheckingAndAssertionOnPageChangeRule in Lombiq.UITestingToolbox

Piedone committed:
Removing just obsolete InstanceCommandLineArgs.Add() in Lombiq.UITestingToolbox

Piedone committed:
Removing just obsolete IsNotFoundMessage() in Lombiq.UITestingToolbox

Piedone committed:
Removing just obsolete BrowserLogMessage and related method in Lombiq.UITestingToolbox

Piedone committed:
Removing Obsolete types/members again, but not IsNotFoundMessage() in Lombiq.UITestingToolbox

Piedone committed:
Dummy change to kick off another build

Piedone committed:
Revert "Dummy change to kick off another build"

This reverts commit 6c55dc632bad683fcdf7b1109a5a31ecfd1495d7.

Piedone committed:
Restoring mistakenly removed IsNotFoundMessage() in Lombiq.UITestingToolbox

Piedone committed:
Dummy change to kick off another build

Piedone committed:
Revert "Dummy change to kick off another build"

This reverts commit 8b4751f194af2d9fb76d033c9bc7478750861dd9.

MZole committed:
Updating Lombiq.GitHub.Actions to latest dev and disabling NuGet tests

MZole committed:
Updating Silenium and WebDriverManager in Lombiq.UITestingToolbox

MZole committed:
Dummy commit to trigger build

MZole committed:
Revert "Dummy commit to trigger build"

This reverts commit 5a1e65e09292ad08585b4c10b1fae1fbdc17a31f.

MZole committed:
Using ReadAllTextAsync in GetLogs() and reverting NLog.config change

MZole committed:
Skipping majority of tests to speed up builds

MZole committed:
Dummy commit to trigger build

MZole committed:
Revert "Dummy commit to trigger build"

This reverts commit 5ece5f34aa1dabf3c6a2bfb33b19fab74021ea04.

MZole committed:
Not excluding IdleTenantsTests and it should fail

MZole committed:
Revert "Not excluding IdleTenantsTests and it should fail"

This reverts commit 8ba67499964bb567eda06edf15bb8c4c07764e36.

MZole committed:
Dummy commit, please fail

MZole committed:
Revert "Dummy commit, please fail"

This reverts commit e5974a213b7a17bee6ce574733a2ae20700e8693.

MZole committed:
Re-adding tests since it didn't fail once with less tests

MZole committed:
Changing NLog.config

MZole committed:
Changing loglevel to Information

MZole committed:
Adding RemoveLoggerFactoryFilter option to nlog to hopefully see logs

MZole committed:
Different appsettings.json config to see if logs appear

MZole committed:
Changing nlog config to see if logs appear

MZole committed:
Dummy commit to kick off another build

MZole committed:
Revert "Dummy commit to kick off another build"

This reverts commit 8f0bd749669d42e5a86078407d4796fcec2a0562.

MZole committed:
Updating Azure.Storage.Blobs version in UITT

MZole committed:
Dummy commit to see if previous build pass is false positive

MZole committed:
Revert "Dummy commit to see if previous build pass is false positive"

This reverts commit 572bce7ca61279f3b086e03fb1eadc17a08ac180.

MZole committed:
Testing if all tests fail when IdleTenantsTests is skipped

MZole committed:
Dummy commit to kick off another build

MZole committed:
Revert "Dummy commit to kick off another build"

This reverts commit f1cd6b7233b929793a3c8aace8d82a570ec5a9c9.

MZole committed:
Setting loglevel to info in NLog.config

MZole committed:
Disabling Html validation on some tests for testing

MZole committed:
Dummy commit to start another build that fails

MZole committed:
Revert "Dummy commit to start another build that fails"

This reverts commit 7c1b277707497cd1f48be00abec17e56e5d89878.

MZole committed:
Reverting Html validation changes from testing

MZole committed:
Adding debug log message for testing in Lombiq.UITestingToolbox

MZole committed:
Dummy commit to kick off another build

MZole committed:
Revert "Dummy commit to kick off another build"

This reverts commit 6854391759ce40e7d9107bb241d1bcc96802494c.

MZole committed:
Placing debug log message in ExecuteTestAfterSetupAsync

MZole committed:
Dummy commit to start another build

MZole committed:
Revert "Dummy commit to start another build"

This reverts commit 5a0f8042910b1ae76e2fe2183506759ddcbc69d2.

MZole committed:
Creating debug point in CreateContextAsync for logging in Lombiq.UITestingToolbox

MZole committed:
Dummy commit to start another build

MZole committed:
Revert "Dummy commit to start another build"

This reverts commit 68f30cf07968b77d9b2f75e3968dbf4dad3158d7.

MZole committed:
More debugging points in  SetupAsync() in Lombiq.UITestingToolbox

MZole committed:
Adding more logging points

MZole committed:
Dummy commit to kick off another build

MZole committed:
Revert "Dummy commit to kick off another build"

This reverts commit 051f8a0ac6c2bcdc4181d4adbc1703667aa4d759.

MZole committed:
Updating UITT and LGHA to latest dev

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Dummy commit to start another build

MZole committed:
Revert "Dummy commit to start another build"

This reverts commit f49bc9fc2842704922163901a20ba7a7fd6e8e1f.

MZole committed:
Changing test config in Lombiq.UITestingToolbox

MZole committed:
Merge branch 'dev' into issue/OSOE-392

MZole committed:
Updating submodules to latest dev branches

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Excluding spelling workflow

MZole committed:
Updating Lombiq.UITestingToolbox to latest dev

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Reverting appsettings.json changes from testing

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Testing with uploading the Temp folder

MZole committed:
Dummy commit to start another build

MZole committed:
Updating upload job in Lombiq.GitHub.Actions

MZole committed:
Moving Temp folder to same location as FailureDumps in Lombiq.GitHub.Actions

MZole committed:
Revereting test changes in Lombiq.UITestingToolbox

MZole committed:
Adding debug log message in Program

MZole committed:
Placing logging outside of Program

MZole committed:
Creating txt file for debugging

MZole committed:
Updating Lombiq.GitHub.Actions to latest dev

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Changing EventId syntax in NLog.config

MZole committed:
Adding the build name to uploaded temp folder

MZole committed:
Dummy commit to start another build

MZole committed:
Updating Lombiq.Hosting.MediaTheme to latest dev

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Dummy commit to start another build

MZole committed:
Revert "Dummy commit to start another build"

This reverts commit dd20e5b273f43c0cf384de9e9f92052dd0aa4d33.

MZole committed:
Starting another build, please fail

MZole committed:
Adding timeout in Lombiq.UITestingToolbox and updating to latest dev Lombiq.HelpfulLibraries

MZole committed:
Updating Lombiq.GitHub.Actions to latest dev

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Post merge fixing in Lombiq.HelpfulLibraries

MZole committed:
Updating submodules to latest dev

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

dministro committed:
Updating submodule:
 - Lombiq.UITestingToolbox: Adding lock around WebApplicationFactory<TEntryPoint>.CreateHost()

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-483

dministro committed:
Temporary removing post-pull-request-checks-automation

dministro committed:
Targeting issue branch in spelling

dministro committed:
Updating submodule:
 - Lombiq.GitHub.Actions: Allowing words

dministro committed:
dummy 1.

dministro committed:
dummy 2.

dministro committed:
dummy 3.

dministro committed:
dummy 4.

dministro committed:
dummy 5.

dministro committed:
Merge branch 'dev' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-392

dministro committed:
Fixing GHA parameter
Updating submodule:
 - Lombiq.UITestingToolbox: Merge

dministro committed:
Merge branch 'issue/OSOE-392' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-483

dministro committed:
Updating submodules:
 - Lombiq.AuditTrailExtensions: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.BaseTheme: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.ChartJs: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.DataTables: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.HelpfulExtensions: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.HelpfulLibraries: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.Hosting.Azure.ApplicationInsights: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.Hosting.Tenants: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.JsonEditor: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.OrchardCoreApiClient: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.Privacy: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.SetupExtensions: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.TrainingDemo: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.UIKit: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
 - Lombiq.VueJs: Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

dministro committed:
Merge branch 'issue/OSOE-392' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-483

dministro committed:
Updating submodules:
 - Lombiq.Gulp.Extensions: fast forward
 - Lombiq.Npm.Targets: fast forward
 - Lombiq.Tests: fast forward
 - Lombiq.Analyzers: fast forward
 - Lombiq.Analyzers.PowerShell: fast forward

dministro committed:
Merge branch 'issue/OSOE-392' of https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions into issue/OSOE-483

dministro committed:
dummy 6.

dministro committed:
dummy 7.

dministro committed:
dummy 8.

dministro committed:
dummy 9.

dministro committed:
dummy 10.

dministro committed:
Addressing GH workflow
Updating submodule:
 - Lombiq.UITestingToolbox: Addressing move OrchardApplicationFactoryCounter below OrchardApplicationFactory<TStartup>

Piedone committed:
Updating Lombiq.UITestingToolbox to merged issue/OSOE-392

[skip ci]

Piedone committed:
Merge pull request #307 from Lombiq/issue/OSOE-483

OSOE-483: Randomly failing UI tests, when upgrading to OC 1.5 in Lombiq.UITestingToolbox

Piedone committed:
Pointing to dev branch

MZole committed:
Reverting changes from debugging

MZole committed:
Updating submodules to latest dev branches

MZole committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

MZole committed:
Reverting more changes from debugging

Piedone commented on NuGetTest/test/Lombiq.OSOCE.NuGet.Tests.UI/Lombiq.OSOCE.NuGet.Tests.UI.csproj:
```diff
@@ -16,16 +16,16 @@
+    <PackageReference Include="Lombiq.Hosting.Tenants.IdleTenantManagement.Tests.UI" Version="2.3.3-alpha.15.osoe-392" />
+    <PackageReference Include="Lombiq.Privacy.Tests.UI" Version="3.2.2-alpha.1.osoe-392" />
+    <PackageReference Include="Lombiq.HelpfulExtensions.Tests.UI" Version="3.6.2-alpha.1.osoe-392" />
+    <PackageReference Include="Lombiq.Tests.UI" Version="4.0.2-alpha.14.osoe-392" />
```
Publish a new alpha of `Lombiq.Tests.UI` with the updates so we can see it working with NuGetTest.

MZole committed:
Updating NuGet package reference of Lombiq.Tests.UI

Piedone committed:
Updating all submodules to merged dev, NuGet publishing

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392

Piedone committed:
Merge remote-tracking branch 'origin/dev' into issue/OSOE-392
